src/app/utils/index.ts

```diff
@@ -1,3 +1,25 @@
+import { useEffect } from "react";
+import api from "../../common/lib/api";
+
 export function classNames(...classes: (string | boolean)[]) {
   return classes.filter(Boolean).join(" ");
 }
+
+export function useThemeEffect(dependencies: undefined | string) {
+  useEffect(() => {
+    api.getSettings().then((response) => {
+      const settings = response;
+      if (settings.theme === "system") {
```

Comment:
can you add a comment here what this is doing?

Reply:
Ok I added some comments to explain how the function works, please let me know if this is what you wanted.